admin-dev/themes/new-theme/js/components/auto-complete-search.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+/**
+ * This component is used with TypeaheadType forms, and is tightly linked to the content of this
+ * twig file src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/typeahead.html.twig
+ *
+ * @todo: the component relies on this TypeaheadType because it was the historic type but it would be worth
+ * creating a new clean form type
+ */
+export default class AutoCompleteSearch {
```

Comment:
I think there is still room for improvement on this one
I'd like it to be more generic, like it only handles search and onSelect call back maybe this it would not be linked to Typeahead any more
Maybe the creation of the wrapping divs should be handed here as well instead of in a twig

Reply:
By wrapping code I mean the div that surrounds the input type which are currently present in this twig template https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/typeahead.html.twig

I don't know typeahead well enough so I don't know if this is needed for it to work or if it's just to store data in attributes and styling purpose

Reply:
But I think for now it's good enough for this PR and can be improved in another PR